src/metabase/search.clj

```diff
@@ -0,0 +1,113 @@
+(ns metabase.search
+  (:require [clojure.core.memoize :as memoize]
+            [clojure.string :as str]
+            [metabase.models.table :refer [Table]]
+            [schema.core :as s]))
+
+;;; Utility functions
+
+(s/defn normalize :- s/Str
+  [query :- s/Str]
+  (str/lower-case query))
+
+(s/defn tokenize :- [s/Str]
+  "Break a search `query` into its constituent tokens"
+  [query :- s/Str]
+  (filter seq
+          (str/split query #"\s+")))
+
+(def ^:private largest-common-subseq-length
+  (memoize/fifo
```

Comment:
the `fifo` here defaults to a queue size of 32. That seems pretty small and perhaps could even run out of useful cached results in the recursions here. I think you can do `(memoize/fif (fn ...) :fifo/threshold 1000)` or perhaps even higher. This seems like useful to keep a cache of these things and fundamentally the data are just `{string -> integer}` so we aren't holding on to a huge amount in the cache.